src/NServiceBus.Core/Transports/Learning/LearningTransportMessagePump.cs

```diff
@@ -104,13 +104,13 @@ await InnerProcessMessages()
 
                     var nativeMessageId = Path.GetFileNameWithoutExtension(filePath);
 
+                    await concurrencyLimiter.WaitAsync(cancellationToken)
```

Comment:
Is there a reason to create the transaction before waiting on the semaphore? Seemed odd to me.